src/common/obj_bencher.cc

```diff
@@ -443,8 +444,9 @@ int ObjBencher::write_bench(int secondsToRun,
   lock.Lock();
   data.finished = 0;
   data.start_time = ceph_clock_now();
+  stopTime = data.start_time + runtime;
   lock.Unlock();
-  for (int i = 0; i<concurrentios; ++i) {
+  for (int i = 0; i<concurrentios && ceph_clock_now() < stopTime; ++i) {
```

Comment:
> - `aio_write()` is a non-blocking call, but if the `concurrentios` is much larger than the variable `<seconds>`, it will not sent completely in the time.  

right. makes sense to me!

> I think the upper module may need to be removed and delete the `contents[slot]` which is not completed. 

we could check `finishTime` in the loop, but if a read op is blocked forever, i don't think there is a simple approach to signal the `wait_for_complete_and_cb()` call.